### DISCUSSION TITLE
Selecting the organization projects

### DISCUSSION THREAD

**Discussion by luyzdeleon** [pokt-network/organization#15] [General]
Team, 

After reviewing everything from the documentation we produced over our 2 weeks of meetings for the Phaseshift process, there's one more detail to finalize: which projects this new organization is going to maintain from our current project list. The idea is to give everyone an open forum on a meeting to have this discussion and understand what needs to be offloaded to 3rd parties via the ecosystem team. I propose we have this meeting on friday because a lot of the transition will hinge on what projects we decide to maintain. Here's an initial list I precompiled:

```
 Pocket Core 
 PocketJS
 Pocket Core Deployment Artifacts (incl. Homebrew)
 Observability Tooling (Grafana Dashboards, Alarms, Etc)
 Validator Nodes
 Seed Nodes
 Beta Nodes
 Testnet Nodes
 Pocket Dashboard
 Pocket Gateway
 Pocket Wallet
 Pocket Block Explorer
 Pocket JS Web3 Provider
 Pocket Testnet Faucet
 MyCrypto Integration
 MyEtherWallet Integration
 Algorand SDK
 PRLTS
 PNS
```

Let's use this thread to discuss the topic asynchronously before the meeting.

**Comment by edjroz** [comment]
I'll propose that we keep the following
```
Pocket Core 
 PocketJS
 Pocket Core Deployment Artifacts (incl. Homebrew)
 Observability Tooling (Grafana Dashboards, Alarms, Etc)
 Validator Nodes
 Seed Nodes
 Testnet Nodes
 Beta Nodes
 PRLTS
 PNS
```

I picked the controversial `PRLTS` because I don't see a way in which protocol specialists won't work without core testing tooling, we could off load this to the community but we probably wanna keep tweaking it or refactor it to match the specific needs of PNI, which could lead to improving functional testing through PRLTS. As for `PNS` I think we could refactor it to run fully automated local setups or remote setups which is an impending necessity as most specialists have their own local setups rather than a uniform lab space to run setups.


I see the remaining
```
Pocket Dashboard
 Pocket Gateway
 Pocket Wallet
 Pocket Block Explorer
 Pocket JS Web3 Provider
 Pocket Testnet Faucet
 MyCrypto Integration
 MyEtherWallet Integration
 Algorand SDK
 ```
 
 all as an ecosystem that can and should be maintained by the community.

**Reply by andrewnguyen22** [reply]
I think we need to redo PRLTS and PNS. PRLTS has a known memory leak and is built in javascript and PNS is very expensive and relatively unoptimized

**Reply by edjroz** [reply]
Right and I won’t disagree with that but think about it in an abstract that we need to work on tooling that helps us improve the protocol, PRLTS as it exists is not the ideal tool but we do need a tool that satisfies that role

**Reply by andrewnguyen22** [reply]
It only satisfies it if you run it with a process manager XD

**Reply by andrewnguyen22** [reply]
Just checked my disk, PRLTS printed 17.5GB worth of logs on my last run

**Reply by luyzdeleon** [reply]
Truth be told, PRLTS exists because of the Dispatch system, we can use something like [https://locust.io/](https://locust.io/) and customize a script that does the Dispatch request.

**Reply by edjroz** [reply]
Locust sounds like a very clever solution actually. 

We still have to replace PNS though, we need a way to setup lab environments remote or local, this is a must for the evolution of the protocol we need to easily plug'n'play different versions of protocols environments etc...

**Reply by luyzdeleon** [reply]
We can combine locust with custom kubernetes clusters to have both nodes and clients.

**Reply by edjroz** [reply]
I don’t disagree with making this a milestone having this on the deployment artifacts could help  the whole community set up environments, and testing mixed versioned setups as well as help our functional testing on clean labs.

**Comment by andrewnguyen22** [comment]
```
Pocket Core 
PocketJS
Pocket Core Deployment Artifacts (incl. Homebrew)
Observability Tooling (Grafana Dashboards, Alarms, Etc)
*Downscaled* Validator Nodes
*Downscaled* Seed Nodes
```

My personal list. No beta or testnet nodes. Testnet is barely even used at this point and is expensive to maintain. If the ecosystem wants the testnet to live again, they will spin up their own nodes. The beta nodes do not make sense in the new `certification framework` (we would also need nodes for RC certification too if we went this route)

I think we need to redo PRLTS and PNS. PRLTS has a known memory leak and is built in javascript and PNS is very expensive and relatively unoptimized

**Reply by edjroz** [reply]
Why Downscale validators? Wouldn’t those work as a plausible income source in the long run ? I agree with downscaling seeds and eliminating beta nodes within the context of the certification framework

**Reply by andrewnguyen22** [reply]
Possibly, but the budget right now is quite high. + one organization having the ability to cause a consensus chain halt is never good (which is currently the state)

**Reply by edjroz** [reply]
the first point doesn't necessarily mean we should reduce.

the second point I might concede, but it brings decentralization questions in place such as if we were to reduce our nodes would any other member of the network have the same power to cause a chain halt ? The way I see it networks grow organically just because you momentarily have the power to halt the network doesn't mean we always will, the more nodes come into place the more decentralized the protocol becomes. However, this takes time, I think that the network would reach an organic inflection point rather than a manufactured one from us giving up nodes.

**Reply by andrewnguyen22** [reply]
Well, I see point 1 counteracting your original point as a `plausible source of income`, when it's hurting us financially now

Point 2: node growth will be organic, but I worry that we might set the bar too high with 76M POKT staked from one organization

**Reply by edjroz** [reply]
🤔 but ultimately if the nodes generate a return its not really a burn as the asset can be liquidated and used as raw revenue. 

I take your point though 76M might be alot.

but one of the reasons we might wanna run a higher number of nodes is to observe a higher number of errors through the distribution of our nodes, just raw statistics, we are more bound to observe errors the higher amount of nodes we have and as an engineer organization I’d say observability of bugs/errors is key and has an intangible value for the organization.

I only bring this point not as a hard push back on scaling down but as a key point on why running nodes is key for us

**Comment by wgarcia4190** [comment]
Pocket Core 
PocketJS
Pocket Core Deployment Artifacts (incl. Homebrew)
Observability Tooling (Grafana Dashboards, Alarms, Etc)
Pocket Dashboard
Pocket Gateway
 Pocket Wallet
 Pocket JS Web3 Provider

In my opinion, these are the products that add the greatest value to the ecosystem. I am not familiar with the PRLTS and PNS terms, and from Andrew's comments, these products need to be worked on.

**Reply by andrewnguyen22** [reply]
A few worries:
1) This is a lot for our organization to take on
2) Our terms of project retention were `core technologies` not `highest value adds`. Remember there is an entire ecosystem team whose sole job is to grow the ecosystem.
3) You could make the argument that we are actually hindering the ecosystem by centralizing all of the tooling to PNI

Interesting choice to have no validators. Wondering if you can go into more detail about this idea.

**Reply by edjroz** [reply]
Can you elaborate on some of your picks ? such as the Dashboard, Gateway Wallet & the Web3 Provider ?

**Comment by aqt01** [comment]
**Core Tech**
 Pocket Core 
 PocketJS
 Pocket Dashboard
 Pocket Gateway
 Pocket Wallet
 Pocket Block Explorer
 
 **Core integrations**
 MyEtherWallet Integration
 
 ** Core infra**
 Validator Nodes (Downscaled)
 Seed Nodes (Downscaled)
 Beta Nodes
 Testnet Nodes (Downscaled)
 
 **Support tools**
 PRLTS
 Pocket Core Deployment Artifacts (incl. Homebrew)
 Observability Tooling (Grafana Dashboards, Alarms, Etc)
 
 
 
 I divided the projects by category in order to have better semantic approach and i will justify them by their category.
 
 I added support tools because as the name refers, we used them to support the development of the core tech. In the case of PRLTS, i think it should be opensourced as the artifcats and observability tools.
 
 Regarding core infra. I think we can downscale lots of our nodes, that includes testnet, so we can operate at a minimum and save that budget spot for more integrations aiming at relay/ecosystem growth
 
  I don't have much context about the other integrations but i think myetherwallet makes sense because is one of the main tools that eth uses and is our major blockchain.
  
 About core tech, when we talk about one blockchain, we also talk about those tools that inevitable the users have their image in mind. Maybe the only one that can be defased is dashboard because we soon no longer we won't be selling pokt by PNI. 
 
 
 
It's important to note that i agreed with a minimal functional maintenace of the tools above mentioned. That means, we should keep at least then working, but not always improving them

**Reply by andrewnguyen22** [reply]
@aqt01 I think this is too much... This is just a small downgrade from what we have now. 
(Taken from my comment to wilson)
1. This is a lot for our organization to take on
2. Our terms of project retention were core technologies not highest value adds. Remember there is an entire ecosystem team whose sole job is to grow the ecosystem.
3. You could make the argument that we are actually hindering the ecosystem by centralizing all of the tooling to PNI

**Comment by oten91** [comment]
My list 

```
 Pocket Core 
 PocketJS
 Pocket JS Web3 Provider
 Pocket Core Deployment Artifacts (incl. Homebrew)
 Observability Tooling (Grafana Dashboards, Alarms, Etc)
 Validator Nodes (downscaled)
 Seed Nodes (max 2)
 PRLTS
 PNS
```
Others seems closely attached to Ecosystem or are products that can evolve on their own.

**Comment by luyzdeleon** [comment]
On the meeting held by the organization on Dec 18th, 2020, the following projects were selected as the organization's owned projects:

```
Pocket core
Pocket JS
Pocket core deployment artifacts
Pocket core observability tooling
Validator nodes *Downscaled*
Seed nodes *Downscaled*
```

Every other project will now be in maintenance mode until delegated to a 3rd party.
